lang/c/src/datafile.c

```diff
@@ -525,7 +525,9 @@ int avro_file_reader_fp(FILE *fp, const char *path, int should_close,
 	r->current_blocklen = 0;
 
 	rval = file_read_block_count(r);
-	if (rval) {
+	if (rval == EOF) {
```

Comment:
Would this handle the datum case as well - avro_file_reader_read? Or as datum is being deprecated does this not need to be handled?